scripts/auth.js

```diff
@@ -0,0 +1,184 @@
+document.addEventListener("DOMContentLoaded", function () {
+  const form = document.querySelector("form.auth-form");
+  if (!form) return;
+  const button = form.querySelector("button.auth-button");
+  button.disabled = true;
+
+  // 전체 폼 유효성 검사
+  function checkFormValidity() {
+    let valid = true;
+    const inputs = form.querySelectorAll("input");
+    inputs.forEach(function (input) {
+      if (input.value.trim() === "") {
+        valid = false;
+      }
+    });
+    if (form.querySelectorAll(".error-message").length > 0) {
+      valid = false;
+    }
+    button.disabled = !valid;
+  }
+
+  // 이메일
+  const emailInput = document.getElementById("email");
+  if (emailInput) {
+    emailInput.addEventListener("blur", validateEmail);
+    emailInput.addEventListener("input", validateEmail);
+  }
+  function validateEmail() {
+    const emailValue = emailInput.value.trim();
+    let errorMessage = emailInput.closest(".input-group").querySelector(".error-message");
+    if (!emailValue) {
+      emailInput.classList.add("input-error");
+      if (!errorMessage) {
+        errorMessage = document.createElement("span");
+        errorMessage.className = "error-message";
+        errorMessage.style.cssText = "color: red; font-size: 14px; display: block; margin-top: 4px;";
+        errorMessage.innerText = "이메일을 입력해주세요.";
+        emailInput.closest(".input-group").appendChild(errorMessage);
```

Comment:
💊 제안
JS 를 통해 스타일을 변경해주셔도 괜찮지만, error와 관련된 클래스를 만들어두시고 해당 클래스를 통해 스타일링하시면
코드도 줄고 css관련은 css 파일에 적으니 더 좋을 것 같네요!

```suggestion
        errorMessage = document.createElement("span");
        errorMessage.className = "error-message"; // css에서 .error-message { color: red; font-size: 14px; display: block; margin-top: 4px; } 와 같은 식으로 스타일링하세요~
        errorMessage.innerText = "이메일을 입력해주세요.";
        emailInput.closest(".input-group").appendChild(errorMessage);
```